### ISSUE TITLE
std::is_constructible not working with aggregates

### ISSUE THREAD

**Issue by JVApen** [llvm/llvm-project#54387]
````
#include <type_traits>

namespace
{
    struct S
    {
        int a{};
        double b{};
    };

    auto f()
    {
        return S{0, 0.};
    }

    static_assert(std::is_constructible_v<S, int, double>);
}
````
https://compiler-explorer.com/z/fqr1r6EjW

Result:
````
<source>:16:5: error: static_assert failed due to requirement 'std::is_constructible_v<(anonymous namespace)::S, int, double>'
    static_assert(std::is_constructible_v<S, int, double>);
    ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
````

Expected: As I can construct S with an int, double, as done in f, I would expect std::is_constructible_v to return `true`.

**Comment by frederick-vs-ja** [comment]
You should use direct-non-list-initialization (`S(0, 0.)`) in your example.

`std::is_constructible_v<S, int, double>` is `true` since C++20 (but is `false` in C++17). However, the clang frontend has not implemented the relative features.
Tracked in #54040, #54041.

**Comment by frederick-vs-ja** [comment]
You should use direct-non-list-initialization (`S(0, 0.)`) in your example.

`std::is_constructible_v<S, int, double>` is `true` since C++20 (but is `false` in C++17). However, the clang frontend has not implemented the relative features.
Tracked in #54040, #54041.

**Comment by JVApen** [comment]
This ain't an issue in libc++, as Ive encountered this with Clang-cl with Microsoft STL. Compiled with MSVC, this behaves as expected.

I'm guessing this is using some built-in from the compiler.

**Comment by alanzhao1** [comment]
Resolved now that #54040 is resolved.
